Sources/MapboxNavigation/PreviewViewController.swift

```diff
@@ -0,0 +1,633 @@
+    }
+    
+    func updateInternalComponents(to state: State) {
+        delegate?.previewViewController(self, stateDidChangeTo: state)
+        
+        switch state {
+        case .browsing:
+            backButton.isHidden = true
+            pointAnnotationManager?.annotations = []
+            navigationView.resumeButton.isHidden = true
+            
+            navigationView.navigationMapView.removeWaypoints()
+            navigationView.navigationMapView.removeRoutes()
+            
+            // Speed limit, road name and floating buttons should be shown.
+            speedLimitView.isAlwaysHidden = false
+            wayNameView.isHidden = false
+            
+            cameraFloatingButton.cameraState = .following
+            
+            if let customBottomBannerViewController = delegate?.previewViewController(self,
+                                                                                      bottomBannerViewControllerFor: state) {
+                navigationView.bottomBannerContainerView.isHidden = false
+                embed(customBottomBannerViewController, in: navigationView.bottomBannerContainerView)
+                
+                presentedBottomBannerViewController = customBottomBannerViewController
+            } else {
+                navigationView.bottomBannerContainerView.hide()
+            }
+        case .destinationPreviewing(let destinationOptions):
+            // Speed limit, road name and floating buttons should be hidden.
+            speedLimitView.isAlwaysHidden = true
+            wayNameView.isHidden = true
+            
+            navigationView.navigationMapView.removeWaypoints()
+            navigationView.navigationMapView.removeRoutes()
+            
+            navigationView.bottomBannerContainerView.show()
+            backButton.isHidden = false
+            
+            navigationView.bottomBannerContainerView.subviews.forEach {
+                $0.removeFromSuperview()
+            }
+            
+            let destinationPreviewViewController: DestinationPreviewing
+            if let customDestinationPreviewViewController = delegate?.previewViewController(self,
+                                                                                            bottomBannerViewControllerFor: state) {
+                guard let customDestinationPreviewViewController = customDestinationPreviewViewController as? DestinationPreviewing else {
+                    preconditionFailure("UIViewController should conform to DestinationPreviewing protocol.")
+                }
+                
+                destinationPreviewViewController = customDestinationPreviewViewController
+            } else {
+                destinationPreviewViewController = DestinationPreviewViewController(destinationOptions)
+                (destinationPreviewViewController as? DestinationPreviewViewController)?.delegate = self
```

Comment:
When we choose a waypoint, and set up the `DestinationPreviewViewController`, the start button should be disabled, because it's not working until we click the `previewButton` to preview. It may appear after the `previewButton` is clicked.